src/yourdfpy/urdf.py

```diff
@@ -1952,6 +1975,15 @@ def _parse_robot(xml_element):
             robot.links.append(URDF._parse_link(l))
         for j in xml_element.findall("joint"):
             robot.joints.append(URDF._parse_joint(j))
+        for m in xml_element.findall("material"):
```

Comment:
Also the `_write_robot` requires for the materials to be written.

Reply:
Done in 7d485d93e727509850af2c1bf8d8993c2d0fe713 :ok_hand: